store/tikv/region_cache.go

```diff
@@ -291,13 +323,18 @@ func (c *RegionCache) getCachedRegion(id RegionVerID) *Region {
 // searchCachedRegion finds a region from cache by key. Like `getCachedRegion`,
 // it should be called with c.mu.RLock(), and the returned Region should not be
 // used after c.mu is RUnlock().
-func (c *RegionCache) searchCachedRegion(key []byte) *Region {
+// If the input key is end key, the second argument is False. for example, use it for reverse scan.
+func (c *RegionCache) searchCachedRegion(key []byte, startKey bool) *Region {
```

Comment:
Could you please rename the second parameter? A boolean value is not a key. Please at lease name it something like `useStartKey` or `isStartKey`.
By the way, in my opinion, it's better to emphasize the difference of it's behavior when you search by end key, so name it `isEndKey` or something.